MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/location/LocationComponentTest.kt

```diff
@@ -386,6 +386,24 @@ class LocationComponentTest {
     verify(compassEngine, never()).addCompassListener(any(CompassListener::class.java))
   }
 
+  @Test
+  fun compass_notAddListenerWhenLayerNotReady() {
+    locationComponent.activateLocationComponent(context, mock(Style::class.java), locationEngine, locationEngineRequest, locationComponentOptions)
+    locationComponent.isLocationComponentEnabled = true
+    locationComponent.onStart()
+    `when`(locationLayerController.isConsumingCompass).thenReturn(true)
+
+    locationComponent.onStartLoadingMap()
+    // Layer should be disabled at this point
+    locationComponent.setCameraMode(CameraMode.TRACKING)
+    verify(compassEngine, never()).addCompassListener(any(CompassListener::class.java))
+
+    locationComponent.onFinishLoadingStyle()
+    // Layer should be enabled at this point
+    locationComponent.setCameraMode(CameraMode.TRACKING_GPS)
```

Comment:
This shouldn't be necessary, the listener should be added automatically when the style is loaded based on the previous data.

Reply:
Yes, this additional verify is added to cover the case where the `isLayerReady` is changed from false to true, the addListener should go through.